frontend/views/containers/chatroom/MessagePoll.vue

```diff
@@ -9,36 +9,41 @@ message-base(
 
   template(#full-width-body='')
     .c-poll-container
-      form.c-poll-form(@submit.prevent='')
-        i18n.c-poll-label(tag='label') poll
-        h3.is-title-4.c-poll-title {{ pollData.question }}
+      component.c-poll-inner(
+        :is='messageContentComponent'
+        :isPollEditable='isPollEditable'
+        :isChangeMode='ephemeral.isChangeMode'
+        :messageId='messageId'
+        :messageHash='messageHash'
+        :pollData='pollData'
+      )
 
-        fieldset.is-column.c-poll-options-list
-          label.c-poll-option(
-            v-for='option in pollData.options'
-            :key='option.id'
-            :class='allowMultipleChoices ? "checkbox" : "radio"'
-          )
-            input.input(v-if='allowMultipleChoices' type='checkbox' :value='option.id' v-model='form.selectedOptions')
-            input.input(v-else type='radio' :name='messageId' :value='option.id' v-model='form.selectedOptions')
-            span.c-poll-option-value {{ option.value }}
-
-        .buttons(v-if='enableSubmitBtn')
-          i18n.is-small(tag='button' type='button' @click='submitSelections') submit
+      banner-scoped(ref='errBanner')
 </template>
 
 <script>
+import sbp from '@sbp/sbp'
+import { mapGetters } from 'vuex'
 import MessageBase from './MessageBase.vue'
-import { MESSAGE_VARIANTS, POLL_TYPES } from '@model/contracts/shared/constants.js'
+import { MESSAGE_VARIANTS, POLL_STATUS } from '@model/contracts/shared/constants.js'
+import { DAYS_MILLIS, MINS_MILLIS } from '@model/contracts/shared/time.js'
+import PollToVote from './poll-message-content/PollToVote.vue'
+import PollVoteResult from './poll-message-content/PollVoteResult.vue'
+import BannerScoped from '@components/banners/BannerScoped.vue'
 
 export default ({
   name: 'MessagePoll',
   components: {
-    MessageBase
+    MessageBase,
+    PollToVote,
+    PollVoteResult,
+    BannerScoped
   },
   props: {
     pollData: Object, // { creator, status, question, options ... }
     messageId: String,
+    messageHash: String,
+    currentUsername: String,
```

Comment:
Is this `currentUsername` used in this component?

Reply:
@Silver-IT Yes, this has to be explicitly specified here so that it's part of `$props` attribute of this component. This was a fix for the issue where the current user's emoticon reaction is not highlighted.